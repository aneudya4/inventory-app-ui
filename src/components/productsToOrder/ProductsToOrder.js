import React from 'react';
import numeral from 'numeral';
import './productsToOrder.css';
const ProductsToOrder = ({ product, quantity, deleteCartProduct }) => {
  const productTotal = parseFloat(product.unit_price) * parseInt(quantity);
  return (
    <div className='orders order-details '>
      <span>{product.product_name}</span>
      <span>{product.unit_price}</span>
      <span>{quantity}</span>
      <span>{numeral(productTotal).format('($0,0)')}</span>
      <span
        className='btn delete'
        onClick={() => deleteCartProduct(product.id)}
      >
        Delete
      </span>
    </div>
  );
};

export default ProductsToOrder;
