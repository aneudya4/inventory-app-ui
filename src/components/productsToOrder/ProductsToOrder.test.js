import { shallow } from 'enzyme';

import ProductsToOrder from './ProductsToOrder.js';

describe('Test on the <ProductsToOrder/> Component', () => {
  it('renders ProductsToOrder without crashing', () => {
    const testProduct = {
      id: 1,
      product_name: 'Iphone 12 Pro',
      user_id: 1,
      unit_price: '799.99',
      description: '64GB',
      stock_total: 18,
      provider: 'Apple',
    };
    const wrapper = shallow(<ProductsToOrder product={testProduct} />);
    // expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
