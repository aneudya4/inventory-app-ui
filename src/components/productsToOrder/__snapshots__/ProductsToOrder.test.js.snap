// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <ProductsToOrder/> Component renders ProductsToOrder without crashing 1`] = `
<div
  className="orders order-details "
>
  <span
    className="btn delete"
    onClick={[Function]}
  >
    <i
      class="fas fa-trash"
    />
     
  </span>
  <span>
    Iphone 12 Pro
  </span>
  <span>
    $799.99
  </span>
  <span />
  <span>
    $NaN
  </span>
</div>
`;
